tests/VirtoCommerce.CatalogModule.Tests/ProductAssociationTests.cs

```diff
@@ -0,0 +1,390 @@
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using VirtoCommerce.CatalogModule.Data.Repositories;
+using VirtoCommerce.CatalogModule.Data.Services;
+using Xunit;
+using dataModel = VirtoCommerce.CatalogModule.Data.Model;
+using MockQueryable.Moq;
+using VirtoCommerce.CatalogModule.Core.Model;
+using VirtoCommerce.Platform.Caching;
+using VirtoCommerce.Platform.Core.Domain;
+
+namespace VirtoCommerce.CatalogModule.Test
+{
+    [Trait("Category", "CI")]
+    public class ProductAssociationTests
+    {
+        Mock<ICatalogRepository> _catalogRepositoryMock = new Mock<ICatalogRepository>();
+
+        public static object[][] ValidEntities => new object[][]
+        {
+            new object[]
+            {
+                new dataModel.AssociationEntity()
+                {
+                    Id = "originalId",
+                    ItemId = "originalItemId",
+                    AssociatedItemId = "originalAssociatedId",
+                    AssociatedCategoryId = null,
+                    AssociationType = "product",
+                    Priority = 0
+                },
+            },
+        };
+
+        [Theory]
+        [MemberData(nameof(ValidEntities))]
+        public async Task AddNotTransientAssociation(dataModel.AssociationEntity entity)
```

Comment:
Please change all test names according to https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices#naming-your-tests, e.g. here

UpdateAssociationsAsync_AddNotTransientAssociation_Added